extension/src/popup/components/signTransaction/Operations/index.tsx

```diff
@@ -300,6 +313,21 @@ export const Operations = ({
     t("Authorization Required; Authorization Required; Authorization Immutable")
   */
 
+  const [contractId, setContractId] = useState("");
+  const [decimals, setDecimals] = useState(0);
+
+  useEffect(() => {
+    const fetchContractDecimals = async () => {
+      const contractDecimals = await getContractDecimals(
```

Comment:
we need to fetch `decimals` to properly show the amount a user is sending. Kind of annoying, but it's probably better to format the amount using the appropriate number of decimals

Reply:
I think one future improvement could be to hold certain high level values related to Soroban tokens in a globally available store for uses like this. Since users have to land on the account tab first when using Freighter, that's a good opportunity to save these off for uses in other workflows in the wallet. Decimals, along with the rest of the token interface can never change so maybe even worth thinking about options for persisting these past a single session.

Reply:
Yeah, definitely! So my original idea was just to grab this from `tokenBalances` in redux, but realized that `getTokenBalances` doesn't run when you just launch `SignTransaction` from freighter-api, we just run that in the `Account` view when you open the extension. There's also the case where a user may want to sign a transaction for a transfer/mint for a token that they haven't loaded into Freighter yet.

So, they would need to `getTokenBalances` to get all their tokens, check to see if they have the one we're looking for, and then possibly do an additional fetch to get the token related to this transaction. Rather than do all that, I thought it'd be more efficient for v1 just to fetch the one decimal we were looking for

But definitely agree that we should come up with something more ergonomic for this